### DISCUSSION TITLE
Preset in private repo within same org

### DISCUSSION THREAD

**Discussion by Xavientois** [renovatebot/renovate#12021] [Request Help] [answered]
While testing the use of Renovatebot within my organization, I first tried it out with a preset defined within a private repo and was successfully able to use it from another one of our private repos. However, now that we are trying to use our preset (stored in a private repo) from one of our public repos, we get `Error type: Cannot find preset's package`.

Is this due to the fact that it is a public repo trying to access the preset of a private repo? Both repos are within the same org and Renovatebot is enabled for both.

Do we need to make the repo containing our renovatebot presets public, or is there another solution?

**Comment by viceice** [comment, marked as answer]
I think the hosted renovate app can't access other private repos. This is by design of the app authentication.

**Reply by Xavientois** [reply]
I see, but if the repo extending the preset is also private, it allows it because having access to one implies having access to the other?

**Reply by viceice** [reply]
No, when both are private, renovate app must be installed in both repos.

**Reply by Xavientois** [reply]
But in my case, the renovate app is installed in both repos, but the repo trying to extend the preset is public. In this case, despite the renovate app being installed in both, it still fails to find the preset's repo.
> I think the hosted renovate app can't access other private repos. This is by design of the app authentication.

Based on your comment, am I correct in saying that, even though the renovate app is installed in both repos, the only way to reference the preset in the private repo from the public repo is to make the private repo public?

**Reply by viceice** [reply]
Yes, if you use the app, both repos need to be public then. A private can access a public repo but a public can't access a private.

**Reply by rarkins** [reply]
FYI this is a hardcoded security feature for now. If we're running in a private repo we say to github "give me one of those tokens that lets me access all repos in the org" but if it's a public repo then we ask for a token scoped only to that repo. The intention is to reduce the chance of leaking private repos from a public repo run by accident.

If you have presets with *some* private credentials, then I'd recommend splitting it into two presets:
- One which is public and can be used by all
- One which contains just the secrets/private info and is used by other private
